### ISSUE TITLE
[NestJS] @nrwl/nestjs@12 not compatible with Angular-Devkit > 11 & npm > 6

### ISSUE THREAD

**Issue by liverant** [nrwl/nx#9588]
## Current Behavior

**TL;DR @nrwl/nestjs@12 should support angular-devkit > 12, and not 11.**

I have an NX monorepo on version 12, the monorepo itself is a node monorepo (no preset) where I have Node & Angular libraries. 
I want to add a NestJs library so it should also be on version 12.
When I try to `npm i -D @nrwl/nest@12.10.1` I'm getting the following error:

```
npm ERR! code EBADENGINE
npm ERR! engine Unsupported engine
npm ERR! engine Not compatible with your version of node/npm: @angular-devkit/core@11.2.4
npm ERR! notsup Not compatible with your version of node/npm: @angular-devkit/core@11.2.4
npm ERR! notsup Required: {"node":">= 10.13.0","npm":"^6.11.0 || ^7.5.6","yarn":">= 1.13.0"}
npm ERR! notsup Actual:   {"npm":"8.3.2","node":"v14.17.5"}
```

If I use the latest `@nrwl/nestjs` I will get this error:

```
npm ERR! code 1
npm ERR! path /home/runner/work/ni-libs/ni-libs/node_modules/@nrwl/js/node_modules/@nrwl/workspace
npm ERR! command failed
npm ERR! command sh -c node ./src/init/init
npm ERR! >  NX   Cannot read properties of undefined (reading 'endsWith')
npm ERR! NX As of Nx 13, project configuration should be moved from nx.json to workspace.json/project.json. Please run "nx format" to fix this.
```

As not all of my NX dependencies are on version 13. 
Upgrading all NX (including Angular) dependencies to version 13 is not possible for me right now.

## Expected Behavior
@nrwl/nest@12 should support angular-devkit > 12, to support npm > 6.

## Steps to Reproduce
packaje.json as follows:

```
{
...
  "dependencies": {
    "@angular/compiler-cli": "^12.2.15",
    "@angular/platform-browser-dynamic": "^12.2.15",
    "ng-packagr": "^12.0.5",
    "tslib": "^2.0.0"
  },
  "devDependencies": {
    "@commitlint/cli": "^16.0.2",
    "@commitlint/config-conventional": "^16.0.0",
    "@next-insurance/eslint-config": "^2.0.1",
    "@nrwl/angular": "^12.10.1",
    "@nrwl/cli": "12.10.0",
    "@nrwl/eslint-plugin-nx": "12.10.0",
    "@nrwl/jest": "12.10.0",
    "@nrwl/linter": "12.10.0",
    "@nrwl/tao": "12.10.0",
    "@nrwl/workspace": "12.10.0",
    "@semantic-release/changelog": "^6.0.1",
    "@semantic-release/exec": "^6.0.3",
    "@semantic-release/git": "^10.0.1",
    "@semantic-release/release-notes-generator": "^10.0.3",
    "@types/jest": "27.0.2",
    "@types/node": "14.14.33",
    "@typescript-eslint/eslint-plugin": "^5.3.1",
    "@typescript-eslint/parser": "^5.3.1",
    "cz-conventional-changelog": "^3.3.0",
    "eslint": "^7.32.0",
    "eslint-config-airbnb-base": "^15.0.0",
    "eslint-config-airbnb-typescript": "^16.1.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-ban": "^1.6.0",
    "eslint-plugin-import": "^2.25.3",
    "eslint-plugin-prettier": "^4.0.0",
    "eslint-plugin-unicorn": "^39.0.0",
    "eslint-plugin-unused-imports": "^2.0.0",
    "husky": "^7.0.4",
    "jest": "27.2.3",
    "jest-preset-angular": "^10.1.0",
    "lint-staged": "^12.1.4",
    "postcss-import": "^14.0.2",
    "prettier": "^2.4.1",
    "semantic-release": "^18.0.1",
    "semantic-release-monorepo": "^7.0.5",
    "ts-jest": "27.0.5",
    "typedoc": "^0.22.10",
    "typescript": "^4.3.5"
  },
  "engines": {
    "npm": ">=7"
  },
...
}
```

and try to install `@nrwl/nestjs@12.10.1`

### Environment
MacOS, Ubuntu CI

**Comment by AgentEnder** [comment]
Hey @liverant, Nx v12 only supports Nest 7. Nest 7's schematics require angular devkit 11.2.4, and there's not anything we can do about that on Nx's side of things. Angular devkit 11.2.4 **_requires_** either npm version 6.11 or 7.5.

You should be able to downgrade your npm version to either of these to proceed. Since this isn't a problem in Nx, I will go ahead and close the issue out. Feel free to let me know if you run into further issues, as we can always reopen it.
